Client panics if no URIs are specified

1. Create a client without specifying any URIs: `client, err := httpclient.NewClient()`
2. Make a request using the client: `_, err = client.Do(context.Background(), httpclient.WithRequestMethod("GET"))`

Expected
---------
Request fails with some kind of error that indicates that no URIs were specified

Actual
------
Panics (specifically, when trying to generate the random offset for the URIs):

```
panic: invalid argument to Intn [recovered]
	panic: invalid argument to Intn

goroutine 20 [running]:
testing.tRunner.func1(0xc00015c100)
	/usr/local/go/src/testing/testing.go:830 +0x392
panic(0x1428f80, 0x155b260)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
math/rand.(*Rand).Intn(0xc0000d0180, 0x0, 0x0)
	/usr/local/go/src/math/rand/rand.go:169 +0x9c
math/rand.Intn(...)
	/usr/local/go/src/math/rand/rand.go:329
github.com/palantir/conjure-go-runtime/conjure-go-client/httpclient.(*clientImpl).Do(0xc000160180, 0x156e360, 0xc0000dc000, 0xc0000cf230, 0x1, 0x1, 0x0, 0x35ff4da001660990, 0x5cdf3f31)
	/Volumes/git/go/src/github.com/palantir/conjure-go-runtime/conjure-go-client/httpclient/client.go:87 +0x6e
github.com/palantir/conjure-go-runtime/conjure-go-client/httpclient_test.TestPanic(0xc00015c100)
	/Volumes/git/go/src/github.com/palantir/conjure-go-runtime/conjure-go-client/httpclient/client_builder_test.go:33 +0x13a
testing.tRunner(0xc00015c100, 0x14ee480)
	/usr/local/go/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x35a
```

--- a/conjure-go-client/httpclient/client.go
+++ b/conjure-go-client/httpclient/client.go
@@ -84,6 +84,9 @@ func (c *clientImpl) Delete(ctx context.Context, params ...RequestParam) (*http.
 
 func (c *clientImpl) Do(ctx context.Context, params ...RequestParam) (*http.Response, error) {
 	uris := c.uris
+	if len(uris) == 0 {
+		return nil, werror.Error("no base URIs are configured")
+	}
 	offset := rand.Intn(len(uris))
 
 	var err error

--- a/conjure-go-client/httpclient/client_test.go
+++ b/conjure-go-client/httpclient/client_test.go
@@ -1,0 +1,32 @@
+// Copyright (c) 2019 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpclient_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/palantir/conjure-go-runtime/conjure-go-client/httpclient"
+)
+
+func TestNoBaseURIs(t *testing.T) {
+	client, err := httpclient.NewClient()
+	require.NoError(t, err)
+
+	_, err = client.Do(context.Background(), httpclient.WithRequestMethod("GET"))
+	require.Error(t, err)
+}

